### PR TITLE
`hypershift destroy cluster openstack`

### PR DESCRIPTION
**Pull request by EmilienM**
```
hypershift destroy cluster openstack --name openstack
2024-07-04T10:49:29-04:00	INFO	Found hosted cluster	{"namespace": "clusters", "name": "openstack"}
2024-07-04T10:49:29-04:00	INFO	Setting client finalizer for hosted cluster	{"namespace": "clusters", "name": "openstack"}
2024-07-04T10:49:29-04:00	INFO	Marking cleanup of cloud resources for hosted cluster	{"namespace": "clusters", "name": "openstack"}
2024-07-04T10:49:29-04:00	INFO	Updated hosted cluster	{"namespace": "clusters", "name": "openstack"}
2024-07-04T10:49:29-04:00	INFO	Deleting hosted cluster	{"namespace": "clusters", "name": "openstack"}
2024-07-04T10:52:01-04:00	INFO	Deleting Secrets	{"namespace": "clusters"}
2024-07-04T10:52:02-04:00	INFO	Deleted CLI generated secrets
2024-07-04T10:52:02-04:00	INFO	Finalized hosted cluster	{"namespace": "clusters", "name": "openstack"}
2024-07-04T10:52:02-04:00	INFO	Successfully destroyed cluster and infrastructure	{"namespace": "clusters", "name": "openstack", "infraID": "openstack"}
~ on ☁️  beaker_openshift(openshift) took 2m33s
```

### CODE REVIEW

**Comment by EmilienM** [comment]
@MaysaMacedo @mdbooth please review

File: cmd/cluster/openstack/destroy.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+		inputErrors = append(inputErrors, fmt.Errorf("infrastructure ID is required"))
+	}
+	if err := errors.NewAggregate(inputErrors); err != nil {
+		return fmt.Errorf("required inputs are missing: %w", err)
+	}
+
+	return core.DestroyCluster(ctx, hostedCluster, o, destroyPlatformSpecifics)
+}
+
+func destroyPlatformSpecifics(ctx context.Context, o *core.DestroyOptions) error {
```
Comment (MaysaMacedo) [inline cmd/cluster/openstack/destroy.go:65]: Isn't there anything specific to delete from OpenStack? or once the HostedCluster is deleted, the openstack cluster also is and consequently OSP resources are wiped out?

**Review by MaysaMacedo — APPROVED** [review]
It looks good to me, I just have one question.

File: cmd/cluster/openstack/destroy.go (hunk 1 above)
Comment (EmilienM) [reply cmd/cluster/openstack/destroy.go:65]: yeah everything is cleaned out, I checked

**Pull request merged**